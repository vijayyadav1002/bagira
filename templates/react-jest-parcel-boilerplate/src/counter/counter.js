import React, { useState } from 'react';

import './counter.scss';

const Counter = () => {
  const [count, setCount] = useState(0);
  setTimeout(() => setCount(count + (count || 1)), 5);

  return (
    <div className='counter'>
      <h1>I will achieve INFINITY</h1>
      <div className='count'>{count}</div>
    </div>
  );
}

export { Counter };