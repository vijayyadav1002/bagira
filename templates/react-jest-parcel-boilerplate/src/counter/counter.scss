.counter {
  width: 100%;
  color: blueviolet;
  font-size: 2rem;
  margin: 20px auto;
  text-align: center;
  border-bottom: 1px solid gold;
  overflow-wrap: break-word;
  .count {
    margin-top: 100px;
    font-size: 10rem;
    color: peru;
    font-weight: 700
  }
}