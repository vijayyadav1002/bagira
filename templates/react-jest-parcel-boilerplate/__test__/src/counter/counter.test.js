import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Counter } from '../../../src/counter/counter';
jest.useFakeTimers();

describe('<Counter/>', () => {
  it('should render correctly', () => {
    const component = shallow(<Counter />);
    expect(setTimeout).toHaveBeenCalled();
    expect(toJson(component)).toMatchSnapshot();
  });
});