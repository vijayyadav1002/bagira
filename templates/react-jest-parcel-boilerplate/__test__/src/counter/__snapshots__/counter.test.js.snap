// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter/> should render correctly 1`] = `
<div
  className="counter"
>
  <h1>
    I will achieve INFINITY
  </h1>
  <div
    className="count"
  >
    0
  </div>
</div>
`;
